llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -1721,6 +1721,10 @@ bool SIFoldOperandsImpl::tryFoldClamp(MachineInstr &MI) {
     return false;
 
   MachineInstr *Def = MRI->getVRegDef(ClampSrc->getReg());
+  MachineInstr *OrigDef = Def;
+  // Look through COPY. COPY only observed with True16.
+  if (Def->isCopy() && Def->getOperand(1).getReg().isVirtual())
+    Def = MRI->getVRegDef(Def->getOperand(1).getReg());
```

Comment:
Can you add MIR tests for this? I have a feeling this copy should have been cleaned up, but you need to defend against undef and subregister uses

Reply:
Upstreamed a MIR test

Reply:
Added. The copy will be replaced to a subreg access if we fixed the copy later and we will come back to this and fix the ll test